android/app/src/main/java/app/organicmaps/car/util/RoutingHelpers.java

```diff
@@ -36,7 +36,7 @@ public static Distance createDistance(@NonNull final app.organicmaps.util.Distan
       break;
     }
 
-    return Distance.create(distance.mDistance, displayUnit);
+    return Distance.create(distance.mDistance < 0 ? 0 : distance.mDistance, displayUnit);
```

Comment:
How it can be negative?

Reply:
![image](https://github.com/organicmaps/organicmaps/assets/10351358/b459038b-5c02-452d-968c-32aa16b18cc3)

Reply:
Does it come from native [platform::distance](https://github.com/organicmaps/organicmaps/blob/a56762c5a6dd2a189507153fc9d39d8d61845efb/platform/distance.hpp)?

https://github.com/organicmaps/organicmaps/blob/a56762c5a6dd2a189507153fc9d39d8d61845efb/android/app/src/main/cpp/app/organicmaps/util/Distance.hpp#L15

I see a method called "isValid()":

https://github.com/organicmaps/organicmaps/blob/a56762c5a6dd2a189507153fc9d39d8d61845efb/platform/distance.cpp#L94

I don't know when distance can be "invalid" in core, but it would be better to fix this invariant at  app.organicmaps.util.Distance level instead of checking in all users.

Reply:
Ok. Distance is `-1` when it's invalid. Distance is invalid when route is recalculated. I added check for it. When route is recalculated we will show loading status.